src/olympia/devhub/tasks.py

```diff
@@ -301,17 +300,6 @@ def handle_upload_validation_result(all_results, upload_pk, channel,
 
     upload = FileUpload.objects.get(pk=upload_pk)
 
-    if waffle.switch_is_active('enable-yara') and results['errors'] == 0:
```

Comment:
Didn't you lose the `results['errors'] == 0` bit ? The `chord()` will still be run when there are errors, right ? And we don't want that...

Reply:
That's a good question. Since `run_yara` is a validation task now, it should not run because `@validation_task` already checks `results['errors']`, no? https://github.com/mozilla/addons-server/blob/4389866635dd83464fd5ad270f080338ea1c645e/src/olympia/devhub/tasks.py#L178-L179

The chord will indeed start but will also early return and the `run_yara` code won't run. Does that sound okay?

If we're not doing this, I don't know how we could do that in the `Validator` since this needs to be decided based on the chain results.

Reply:
Ah yeah forgot we had this check in `@validation_task` now. That's ok. Maybe worth adding an integration test specifically checking that this particular task is aborting early (I know we have the test for the `validation_task` decorator itself, but this is to be extra paranoid about it)